res/controllers/Denon-Prime-4-scripts.js

```diff
@@ -0,0 +1,921 @@
+    components.Deck.call(this, deckNumbers);
+    const theDeck = this;
+
+    // Censor Button
+    this.censorButton = new components.Button({
+        midi: [0x90 + midiChannel, 0x01],
+        key: "reverseroll",
+        unshift: function() {
+            this.inKey = "reverseroll";
+        },
+        shift: function() {
+            this.inKey = "reverse";
+        },
+    });
+
+    // Skip Backward
+    this.skipBackButton = new components.Button({
+        midi: [0x90 + midiChannel, 0x04],
+        key: trackSkipMode[0],
+    });
+
+    // Skip Forward
+    this.skipFwdButton = new components.Button({
+        midi: [0x90 + midiChannel, 0x05],
+        key: trackSkipMode[1],
+    });
+
+    // Beatjump Backward
+    this.bjumpBackButton = new components.Button({
+        midi: [0x90 + midiChannel, 0x06],
+        key: "beatjump_backward",
+    });
+
+    // Beatjump Forward
+    this.bjumpFwdButton = new components.Button({
+        midi: [0x90 + midiChannel, 0x07],
+        key: "beatjump_forward",
+    });
+
+    // Sync Button
+    this.syncButton = new components.SyncButton({
+        midi: [0x90 + midiChannel, 0x08],
+    });
+
+    // Cue Button
+    this.cueButton = new components.CueButton({
+        midi: [0x90 + midiChannel, 0x09],
+    });
+
+    // Play Button
+    this.playButton = new components.PlayButton({
+        midi: [0x90 + midiChannel, 0x0A],
+        unshift: function() {
+            components.PlayButton.prototype.unshift.call(this);
+            this.type = components.Button.prototype.types.toggle;
+        },
+        shift: function() {
+            this.inKey = "play_stutter";
+            this.type = components.Button.prototype.types.push;
+        }
+    });
+
+    // Performance Pads
+    this.padGrid = new Prime4.PadSection(this, midiChannel - 4);
+
+    // Tempo Fader
+    this.tempoFader = new components.Pot({
+        midi: [0xB0 + midiChannel, 0x1F],
+        inKey: "rate",
+        invert: true,
+    });
+
+    // LED indicator when pitch fader is at centre.
+    this.tempoFeedback = new components.Button({
+        midi: [0x90 + midiChannel, 0x34],
+        outKey: "rate",
+        on: 0x02,
+        off: 0x00,
+        outValueScale: function(value) {
+            return value === 0 ? this.on : this.off;
+        },
+    });
+
+    // Keylock Button
+    this.keylockButton = new components.Button({
+        midi: [0x90 + midiChannel, 0x22],
+        key: "keylock",
+        type: components.Button.prototype.types.toggle,
+    });
+
+    this.vinylButton = new components.Button({
+        midi: [0x90 + midiChannel, 0x23],
+        type: components.Button.prototype.types.toggle,
+        input: function(channel, control, value, status, _group) {
+            if (!this.isPress(channel, control, value, status)) {
+                return;
+            }
+            theDeck.jogWheel.vinylMode = !theDeck.jogWheel.vinylMode;
+            this.output(theDeck.jogWheel.vinylMode);
```

Comment:
keeps the code [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)er.
```suggestion
            this.trigger();
```